js/src/logger.ts

```diff
@@ -1561,16 +1792,24 @@ class BackgroundLogger {
       return;
     }
     try {
-      const allItems = await this.unwrapLazyValues(wrappedItems);
+      const [allItems, allAttachments] =
+        await this.unwrapLazyValues(wrappedItems);
       const dataStr = constructLogs3Data(
         allItems.map((x) => JSON.stringify(x)),
       );
+      const attachmentStr = JSON.stringify(
+        allAttachments.map((a) => a.debugInfo()),
+      );
+
+      const promises: Promise<void>[] = [];
       for (const payloadDir of publishPayloadsDir) {
-        await BackgroundLogger.writePayloadToDir({
-          payloadDir,
-          payload: dataStr,
-        });
+        const write = (payload: string) =>
+          BackgroundLogger.writePayloadToDir({ payloadDir, payload });
+        promises.push(write(dataStr));
+        promises.push(write(attachmentStr));
       }
+
+      await Promise.allSettled(promises);
```

Comment:
I think `Promise.all` is fine here since this way won't propagate any errors from running the promises. Writing the debug info is a best-effort thing anyhow. We could also just serially await the promises since this whole thing happens in the background.

Reply:
Done, restored serial code